crates/toolchain/build/src/lib.rs

```diff
@@ -377,6 +377,12 @@ pub fn find_unique_executable<P: AsRef<Path>, Q: AsRef<Path>>(
         .targets
         .into_iter()
         .filter(move |target| {
+            // always filter out build script target
+            if target.kind.iter().any(|k| k == "custom-build")
+                && target.name == "build-script-build"
```

Comment:
isn't this custom to the name of your specific crate/package?

Reply:
I don't think so, all build script should have this name.

Reply:
ok I'm convinced: https://github.com/rust-lang/cargo/blob/611b7c4ee40b63d0818040f7290a8e14424a9519/src/cargo/core/compiler/layout.rs#L70